tdx-attestation-app/src/HttpClient.cpp

```diff
@@ -0,0 +1,147 @@
+//-------------------------------------------------------------------------------------------------
+// <copyright file="HttpClient.cpp" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation.  All rights reserved.
+// </copyright>
+//-------------------------------------------------------------------------------------------------
+
+#include <curl/curl.h>
+#include <json/json.h>
+#include <math.h>
+#include <fstream>
+#include <iostream>
+#include <unordered_map>
+#include <stdio.h>
+#include "HttpClient.h"
+
+#define HTTP_STATUS_OK 200
+#define HTTP_STATUS_BAD_REQUEST 400
+#define HTTP_STATUS_RESOURCE_NOT_FOUND 404
+#define HTTP_STATUS_TOO_MANY_REQUESTS 429
+#define HTTP_STATUS_INTERNAL_SERVER_ERROR 500
+#define MAX_RETRIES 3
+
+using namespace std;
+
+HttpClientResult HttpClient::InvokeHttpRequest(std::string& http_response,
+                                                const std::string& url,
+                                                const HttpClient_I::HttpVerb& http_verb,
+                                                const std::vector<std::string>& header_list,
+                                                const std::string& request_body) {
+
+    curl_global_cleanup();
+
+    CURL* curl_handle;
+    curl_handle = curl_easy_init();
+    if (curl_handle == nullptr) {
+        fprintf(stderr, "Failed to initialize curl_handle for http request.");
+        return HttpClientResult::FAILED;
```

Comment:
We need to call curl_global_cleanup() after curl_easy_init().  I suggest we add a :exit label at the end of the function have a goto statement to that label.

Reply:
I add it at the beginning of this call to ensure that its cleanup before doing anything. This is to avoid a lot of clanup for each scenario